microsetta_interface/implementation.py

```diff
@@ -2480,12 +2522,22 @@ def admin_barcode_search_query_qiita(body):
 
 
 def get_ajax_check_kit_valid(kit_name):
+    if re.match(
+            r"[dD][mM][kK][234689ACDEFHJKMNPRTVWXYacdefhjkmnprtvwxy]{6}",
+            kit_name
+    ):
+        kit_name = kit_name.upper()
     kit, error, _ = _get_kit(kit_name)
     result = True if error is None else error
     return flask.jsonify(result)
 
 
 def get_ajax_list_kit_samples(kit_name):
+    if re.match(
+            r"[dD][mM][kK][234689ACDEFHJKMNPRTVWXYacdefhjkmnprtvwxy]{6}",
```

Comment:
It would be good to define the regex once, maybe at global level, eg:

```python
DAKLAPACK_KIT_RE = re.compile(r"...")
```

Reply:
Adjusting.